config-object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedFilterPushedDownObjectStore.java

```diff
@@ -71,14 +72,41 @@ public Optional<T> getMatchingData(RequestContext context, F filterInput, List<S
     return getMatchingObject(context, filterInput, sortInput).map(ConfigObject::getData);
   }
 
+  public ConfigsResponse<T> getMatchingObjectsWithTotalCount(
+      RequestContext context, F filterInput, List<S> sortInput, Pagination pagination) {
+    List<SortBy> sortByList = sortInput.stream().map(this::buildSort).collect(Collectors.toList());
```

Comment:
nit - can we refactor this to share the implementation with `getMatchingObjects`? other than sending the flag/reading the total on response, they're identical.